### ISSUE TITLE
Can we use TabularPredictor by .fit(X, y) ?

### ISSUE THREAD

**Issue by jckkvs** [awslabs/autogluon#1479]
In the document,
TabularPredictor (label = "class"). fit (train_data)

However, I want to use it like models of sklearn.

predictor = TabularPredictor ()
predictor.fit (X, y)

Is it possible? 

I want to use AutoGluon with sklearn.pipeline, but I think it requires fit(X,y).

**Comment by willsmithorg** [comment]
Right now I don't think AutoGluon has a sci-kit learn compatible API.

You could wrap AutoGluon in a simple layer that took X and y and did `combined_df = pd.concat(X, y, axis=1)` then passed `combined_df` to autogluon.fit() with `label=y.columns[0]`

**Comment by Innixma** [comment]
We may consider adding a sklearn compatible API in future, but haven't dived into all the requirements necessary to make that happen yet.

**Comment by jckkvs** [comment]
@willsmithorg
@Innixma 

Thanks. I try to make the wrapper for sklearn.

By the way, isn't AutoGluon compatible with multiple objective variables? For example, scikit-learn's MultioutputRegressor, MultiOutputClassifier （編集済み）元のテキストに戻す

**Comment by Innixma** [comment]
@jckkvs 
> is AutoGluon compatible with multiple objective variables? Like scikit-learn's MultioutputRegressor, MultiOutputClassifier

AutoGluon can do this, but requires custom code, refer to this tutorial:

https://auto.gluon.ai/stable/tutorials/tabular_prediction/tabular-multilabel.html

**Comment by Innixma** [comment]
I've created #1493 to track the addition of a sklearn compatible API. @jckkvs if you find success in wrapping AG for sklearn, it would be great if you could mention details in that issue.

**Comment by jckkvs** [comment]
@Innixma 
Thanks.

Here is the code.
Its main purpose is to be able to be used with sklearn's Pipeline, MultiOutputRegressor and (GridSearchCV). 

```
import numpy as np
import pandas as pd
import autogluon as ag
from autogluon.tabular  import TabularPredictor

class AutoGluonPredictor(TabularPredictor):
    def __init__(self,
                 label='',
                 problem_type=None,
                 eval_metric=None,
                 path=None,
                 verbosity=0,
                 sample_weight=None,
                 weight_evaluation=False,
                 groups=None,
                 presets='best_quality'):
        super().__init__(label=label,
                         problem_type=problem_type,
                         eval_metric=eval_metric,
                         path=path,
                         verbosity=verbosity,
                         sample_weight=sample_weight,
                         weight_evaluation=weight_evaluation,
                         groups=groups)
        
        self.presets = presets

    def fit(self, X, y=None, sample_weight=None, check_input=True):
        X_df = pd.DataFrame(X)
        y_df = pd.DataFrame(y)
        X_names = X_df.columns
        y_names = y_df.columns
        # This code has not been well considered, but is to resolve some errors. 
        if len(X_names) != len(set(X_names)):
            X_names = [f'X{i}' for i in range(len(X_names))]
        else:
            if len(list(set(X_names) & set(y_names))) >= 1 :
                X_names = [f'X{i}' for i in range(len(X_names))]
                y_names = ['y0']

        X_df.columns = X_names
        y_df.columns = y_names
        y_name = y_names[0]
        self.X_names = X_names
        self.y_names = y_names
        ###########

        self.__init__(label=y_name)
        train_data = pd.concat([X_df, y_df], axis=1)
        super().fit(train_data, presets=self.presets)    
        return self

    def predict(self, X, y=None, sample_weight=None, check_input=True):
        X_df = pd.DataFrame(X)
        X_df.columns = self.X_names

        return super().predict(X_df)

    def set_params(self, **parameters):
        for parameter, value in parameters.items():
            setattr(self, parameter, value)
        return self

    def get_params(self, deep=True):
        """
        Parameters
        ----------
        deep : Ignored. (for compatibility with sklearn)
        Returns
        ----------
        self : returns an dictionary of parameters.
        """

        params = {}
        return params        

```

I'm not thinking of using GridSearchCV for AutoGluon, but I have defined set_params to describe it in the same notation as other estimators (like Lasso). 

Also, in my special usage, I sometimes connect variable selection with Pipeline before AutoGluon, so I added special processing in fit for X_names and y_names. If you don't need it, please ignore it. 


The usage example is described below. 

```
estimator = AutoGluonPredictor()
estimator = MultiOutputRegressor(estimator)
param_grid = {}
model_name = 'AutoGluon'

from sklearn.pipeline import Pipeline
pipe = Pipeline([('selector', SelectFromModel(Lasso()) ), ('estimator', estimator)])
gsv = GridSearchCV(pipe, param_grid= {}) # No-tuning

otherpipe = Pipeline([('selector', SelectFromModel(Lasso()) ), ('estimator', Lasso())])
othergsv =  GridSearchCV(otherpipe , param_grid= {'alpha':[0.1,0.01]}) 


```
